sql-plugin/src/main/311+-nondb/scala/com/nvidia/spark/rapids/shims/v2/GpuBroadcastHashJoinExec.scala

```diff
@@ -42,16 +43,60 @@ class GpuBroadcastHashJoinMeta(
     join.leftKeys.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
   val rightKeys: Seq[BaseExprMeta[_]] =
     join.rightKeys.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
-  val condition: Option[BaseExprMeta[_]] =
+  val conditionMeta: Option[BaseExprMeta[_]] =
     join.condition.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
 
+  def isAstCondition(): Boolean = {
+    join.condition.isDefined && conditionMeta.forall(_.canThisBeAst)
+  }
+
   override val namedChildExprs: Map[String, Seq[BaseExprMeta[_]]] =
-    JoinTypeChecks.equiJoinMeta(leftKeys, rightKeys, condition)
+    JoinTypeChecks.equiJoinMeta(leftKeys, rightKeys, conditionMeta)
 
-  override val childExprs: Seq[BaseExprMeta[_]] = leftKeys ++ rightKeys ++ condition
+  override val childExprs: Seq[BaseExprMeta[_]] = leftKeys ++ rightKeys ++ conditionMeta
 
   override def tagPlanForGpu(): Unit = {
-    GpuHashJoin.tagJoin(this, join.joinType, join.leftKeys, join.rightKeys, join.condition)
+    //GpuHashJoin.tagJoin(this, join.joinType, join.leftKeys, join.rightKeys, join.condition)
+    val joinType = join.joinType
+    val keyDataTypes = (leftKeys ++ rightKeys).map(_.dataType)
+    val gpuBuildSide = GpuJoinUtils.getGpuBuildSide(join.buildSide)
+
+    def unSupportNonEqualCondition(): Unit = if (join.condition.isDefined) {
+      this.willNotWorkOnGpu(s"$joinType joins currently do not support conditions")
+    }
+    def unSupportNonEqualNonAst(): Unit = if (join.condition.isDefined) {
+      conditionMeta.foreach(requireAstForGpuOn)
+    }
+    def unSupportStructKeys(): Unit = if (keyDataTypes.exists(_.isInstanceOf[StructType])) {
+      this.willNotWorkOnGpu(s"$joinType joins currently do not support with struct keys")
+    }
+
+    JoinTypeChecks.tagForGpu(joinType, this)
+    joinType match {
+      case _: InnerLike =>
+      case RightOuter | LeftOuter | LeftSemi | LeftAnti =>
+        unSupportNonEqualNonAst()
```

Comment:
I vote for the `unSupport*` methods to be just part of the case statement.

Reply:
For this commit, I was trying to minimize diffs with GpuHashJoin.tagJoin, in the hopes that once I implement the other two joins we can use the common method for all of them again.  So I will hold off on making this change for now until I can make it in the common method.

Reply:
Thanks for the pointer @jbrennan333, agree with keeping the diffs minimal with that existing function.